ConfigSpace/hyperparameters.py

```diff
@@ -239,85 +150,56 @@ class UnParametrizedHyperparameter(Constant):
 
 
 class NumericalHyperparameter(Hyperparameter):
-    # todo : type of name and default?
     def __init__(self, name: str, default: Any) -> None:
         super(NumericalHyperparameter, self).__init__(name)
         self.default = default
 
     def has_neighbors(self) -> bool:
         return True
 
-    def get_num_neighbors(self, value: None=None) -> np.inf:
+    def get_num_neighbors(self, value= None) -> np.inf:
         return np.inf
 
 
 class FloatHyperparameter(NumericalHyperparameter):
-    # todo : type of name and default?
     def __init__(self, name: str, default: Union[int, float]) -> None:
         super(FloatHyperparameter, self).__init__(name, default)
 
     @abstractmethod
     def is_legal(self, value: Union[int, float]) -> bool:
-        # return isinstance(value, float) or isinstance(value, int)
         raise NotImplemented
 
-    # todo : recheck default
+    @abstractmethod
     def check_default(self, default: Union[int, float]) -> float:
-        # return np.round(float(default), 10)
         raise NotImplemented
 
+
 class IntegerHyperparameter(NumericalHyperparameter):
-    # todo : type of name and default?
     def __init__(self, name: str, default: int) -> None:
         super(IntegerHyperparameter, self).__init__(name, default)
 
     @abstractmethod
     def is_legal(self, value: int) -> bool:
         raise NotImplemented
 
-    def check_int(self, parameter: int, name: str) -> int:
-        raise NotImplemented
-
-    def check_default(self, default: int) -> int:
+    @abstractmethod
+    def check_default(self, default) -> int:
```

Comment:
this is abstract method. in its definition in UniformIntegerHyperparameter type of default is 'Union[int, float]' while in NormalIntegerHyperparameter type of default is 'int' so because of this conflict i removed it form abstract method.

if you want in some different way i can do that for example making it same and then in __init__ of NormalIntegerHyperparameter doing a check to make sure that its only 'int'